packages/flutter/lib/src/painting/image_provider.dart

```diff
@@ -991,17 +991,19 @@ class FileImage extends ImageProvider<FileImage> {
   Future<ui.Codec> _loadAsync(FileImage key, DecoderBufferCallback? decode, DecoderCallback? decodeDeprecated) async {
     assert(key == this);
 
-    final Uint8List bytes = await file.readAsBytes();
-    if (bytes.lengthInBytes == 0) {
+    final int lengthInBytes = file.lengthSync();
+    if (lengthInBytes == 0) {
       // The file may become available later.
       PaintingBinding.instance.imageCache.evict(key);
       throw StateError('$file is empty and cannot be loaded as an image.');
     }
-
     if (decode != null) {
-      return decode(await ui.ImmutableBuffer.fromUint8List(bytes));
+      if (file.runtimeType == File) {
```

Comment:
Is it possible to have test coverage of the `!= File` case?

Reply:
Lots of the existing tests use a MemoryFileSystem entity, so they would break without this change